src/parse.ts

```diff
@@ -116,7 +117,7 @@ export function parseAssetFilename(filename: string): SanityAssetIdParts {
  * @throws If URL is invalid or not a Sanity asset URL
  */
 export function parseAssetUrl(url: string): SanityAssetUrlParts {
-  if (!cdnUrlPattern.test(url)) {
+  if (!(cdnUrlPattern.test(url) || customCdnUrlPattern.test(url))) {
```

Comment:
On second read, I think these would read better as:
```suggestion
  if (!cdnUrlPattern.test(url) && !customCdnUrlPattern.test(url)) {
```

Reply:
agreed! I think I was trying to over-optimize here -- but reading again, I realize that, since we need to evaluate if BOTH parts of the inner parenthesis, there's actually no way to skip the second check (which only adds a few fractions of an MS   anyway). I'll add a commit.